costmap_2d/src/footprint.cpp

```diff
@@ -66,6 +68,130 @@ void calculateMinAndMaxDistances(const std::vector<geometry_msgs::Point>& footpr
   max_dist = std::max(max_dist, std::max(vertex_dist, edge_dist));
 }
 
+BoundingRect minBoundingRect(const std::vector<geometry_msgs::Point>& points)
```

Comment:
Q: did you come up with this implementation by yourself? Can you link useful references you followed? Some general description of how this function works would help me a lot to review this code.

Reply:
I added the reference to the header comment